code/modules/surgery/organs/eyes.dm

```diff
@@ -140,3 +139,189 @@
+
+/obj/item/organ/eyes/robotic/glow/Destroy()
+	terminate_effects()
+	..()
+
+/obj/item/organ/eyes/robotic/glow/Remove()
+	terminate_effects()
+	..()
+
+/obj/item/organ/eyes/robotic/glow/proc/terminate_effects()
+	if(owner && active)
+		deactivate()
+	clear_visuals(TRUE)
+	STOP_PROCESSING(SSfastprocess, src)
+
+/obj/item/organ/eyes/robotic/glow/process()
+	if(owner && (olddir != owner.dir) && active)
+		update_visuals()
+		olddir = owner.dir
+
+/obj/item/organ/eyes/robotic/glow/ui_action_click(owner, action)
+	if(istype(action, /datum/action/item_action/organ_action/toggle))
+		toggle_active()
+	else if(istype(action, /datum/action/item_action/organ_action/use))
+		prompt_for_controls(owner)
+
+/obj/item/organ/eyes/robotic/glow/proc/toggle_active()
+	active ? deactivate() : activate()
+
+/obj/item/organ/eyes/robotic/glow/proc/prompt_for_controls(owner)
+	var/C = input(owner, "Select Color", "Select color", "#ffffff") as null|color
+	if(!C)
+		return
+	var/range = input(owner, "Enter range (0 - [max_light_beam_distance])", "Range Select", 0) as null|num
+	if(!range)
+		return
+
+	set_distance(Clamp(range, 0, max_light_beam_distance))
+	assume_rgb(C)
+
+/obj/item/organ/eyes/robotic/glow/proc/assume_rgb(newcolor)
+	current_color_string = newcolor
+	eye_color = RGB2EYECOLORSTRING(current_color_string)
+	sync_light_effects()
+	cycle_mob_overlay()
+	if(owner)
+		owner.regenerate_icons()
+
+/obj/item/organ/eyes/robotic/glow/proc/cycle_mob_overlay()
+	remove_mob_overlay()
+	sync_mob_overlay()
+	add_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/sync_mob_overlay()
+	mob_overlay.color = current_color_string
+
+/obj/item/organ/eyes/robotic/glow/proc/add_mob_overlay()
+	if(owner)
+		owner.add_overlay(mob_overlay)
+
+/obj/item/organ/eyes/robotic/glow/proc/remove_mob_overlay()
+	if(owner)
+		owner.cut_overlay(mob_overlay)
+
+/obj/item/organ/eyes/robotic/glow/emp_act()
+	if(active)
+		deactivate(silent = TRUE)
+
+/obj/item/organ/eyes/robotic/glow/on_mob_move()
+	if(active)
+		update_visuals()
+
+/obj/item/organ/eyes/robotic/glow/proc/activate(silent = FALSE)
+	start_visuals()
+	if(!silent)
+		to_chat(owner, "<span class='warning'>Your [src] clicks and makes a whining noise, before shooting out a beam of light!</span>")
+	active = TRUE
+	cycle_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/deactivate(silent = FALSE)
+	clear_visuals()
+	if(!silent)
+		to_chat(owner, "<span class='warning'>Your [src] shuts off!</span>")
+	active = FALSE
+	remove_mob_overlay()
+
+/obj/item/organ/eyes/robotic/glow/proc/update_visuals()
+	if((eye_lighting.len < light_beam_distance) || !on_mob)
+		regenerate_light_effects()
+	var/turf/scanfrom = get_turf(owner)
+	var/scandir = owner.dir
+	if(!istype(scanfrom))
+		clear_visuals()
+	var/turf/scanning = scanfrom
+	var/stop = FALSE
+	on_mob.forceMove(scanfrom)
+	for(var/i in 1 to light_beam_distance)
+		scanning = get_step(scanning, scandir)
+		if(scanning.opacity)
```

Comment:
if(scanning.opacity || scanning.has_opaque_atom) and then you can remove the loop